[feat] 공연 검색 기능 구현

wndgndi opened the pull request:
## 📝작업 내용

 - 공연 검색 기능 구현


## 💬리뷰 참고사항
![image](https://github.com/HalfGallonTeam/WithCon_BE/assets/102509248/be1f9ce1-67b1-49e6-b935-4b040e43cffc)

- 공연 배우 명단 혹은 공연 제목에 키워드가 들어간 모든 공연을 조회합니다.
- 공연 배우는 공연 세부 정보 테이블의 필드이고, 공연 제목은 공연 테이블의 필드입니다.
- 공연 세부 정보 엔티티와 공연 엔티티는 동일한 Id를 사용합니다. 
- 반환 타입이 PerformanceResponse이기에 공연 테이블에서 제목에 키워드가 포함되는 공연을 찾을 순 있지만, 공연 세부 정보에서 찾았을 때는 Id를 통해 DB를 한번 더 조회해서 공연을 찾아야 합니다.
- 따라서 한 번의 접근으로 필요한 데이터를 가져오는 것이 효율적이라 생각하여 QueryDSL을 사용했습니다.


(close) #75

wndgndi committed:
feat: QueryDSL 사용하기 위한 Custom Repostiry 생성

공연을 조회할 때 키워드를 사용하여 키워드가 공연명 혹은 출연자에 포함된 모든 공연을 검색합니다.
공연명과 출연자가 다른 테이블의 필드이기에 한번의 쿼리로 데이터를 가져오기 위해 QueryDSL을 사용했습니다.

wndgndi committed:
refactor: 공연 응답 id 필드 제거

id는 필드는 굳이 응답값에 없어도 된다고 생각하여 제거했습니다.

wndgndi committed:
feat: 공연 검색 기능 구현

키워드를 사용하여 해당 키워드가 포함된 모든 공연을 검색할 수 있습니다.

wndgndi committed:
refactor: 공연 응답값에 id 추가

wndgndi committed:
test: 공연 검색 기능 테스트 코드 구현

kdmin0706 reviewed (approved):
수고하셨습니다~

cchoijjinyoung reviewed (approved):
수고하셨습니다! 페이지네이션은 없이 List로 리턴하시는 것으로 확인하였습니다. 이유를 여쭤봐도 될까요?

wndgndi commented:
@cchoijjinyoung 
일단 API 명세서에 작성된 대로 만들어봤습니다.
프론트엔드분들과 이야기해보고 수정이 필요하다면 수정하려 합니다!

sosa7753 reviewed (approved):
수고하셨습니다!

wndgndi committed:
refactor: List 타입에서 Page타입으로 반환타입 변경

wndgndi committed:
refactor: 공연 dto 수정

- PerformanceRequest에 likes는 요청값에 포함될 필요가 없다고 생각하여 제거

- PerformanceResponse에 id값 다시 추가
